src/Session.java

```diff
@@ -1,72 +1,128 @@
 
-            guessLoop: while (true) {
-                String guess = input.nextLine();
+      System.out.println(currentPlayer + ": please enter a guess");
 
-                switch (word.guess(guess)) {
-                    case CORRECT_GUESS:
-                        System.out.println("Correct guess!");
-                        break guessLoop;
-                    case INCORRECT_GUESS:
-                        lives--;
-                        System.out.println("Incorrect guess, lose a live");
-                        break guessLoop;
-                    case UNGUESSABLE_CHAR:
-                        System.out.println("Character cannot be guessed, please pick another");
-                        break;
-                    case COMPLETED_WORD:
-                        players.forEach(p -> p.incrementScore(5));
-                        currentPlayer.incrementScore(5);
-                        leader.decrementScore(3);
-                        System.out.println(word);
-                        System.out.println("Game won! 10 points to " + currentPlayer + ", 5 to others, leader loses 3");
-                        return;
-                    case CHAR_ALREADY_GUESSED:
-                        System.out.println("Character already guessed, please pick another");
-                        break;
-                }
-            }
+      guessLoop:
+      while (true) {
+        String guess = input.nextLine();
 
+        switch (word.guess(guess)) {
+          case CORRECT_GUESS:
+            System.out.println("Correct guess!");
+            break guessLoop;
+          case INCORRECT_GUESS:
+            lives--;
+            System.out.println("Incorrect guess, lose a live");
+            leaderBoard = updateLeaderBoard();
+            break guessLoop;
+          case UNGUESSABLE_CHAR:
+            System.out.println("Character cannot be guessed, please pick another");
+            break;
+          case COMPLETED_WORD:
+            players.forEach(p -> p.incrementScore(5));
+            currentPlayer.incrementScore(5);
+            leader.decrementScore(3);
             System.out.println(word);
-            currentPlayerIndex = (currentPlayerIndex + 1) % players.size();
+            System.out.println(
+                "Game won! 10 points to " + currentPlayer + ", 5 to others, leader loses 3");
+            return;
+          case CHAR_ALREADY_GUESSED:
+            System.out.println("Character already guessed, please pick another");
+            break;
         }
+      }
 
-        System.out.println("No lives left, round lost");
-        leader.incrementScore(10);
+      System.out.println(word);
+      System.out.println("LeaderBoard:");
+      System.out.println(leaderBoard);
+
+      currentPlayerIndex = (currentPlayerIndex + 1) % players.size();
     }
 
+    System.out.println("No lives left, round lost");
+    System.out.println(word);
+    leader.incrementScore(10);
+  }
 
+  private String updateLeaderBoard() {
```

Comment:
This is not a leaderboard. This is a graphic of a hanging man, being shown according to the number of lives you have left.

You're also not "updating" it, as no outside state is being touched; you are simply generating a String and returning it, so it should be more of a "get" than an update.